test/parallel/test-http-many-ended-pipelines.js

```diff
@@ -51,6 +52,13 @@ const server = http.createServer(function(req, res) {
 server.listen(0, function() {
   const client = net.connect({ port: this.address().port,
                                allowHalfOpen: true });
+
+  client.on('error', function(err) {
+    // The socket might be destroyed by the other peer while data is still
+    // being written.
+    assert.strictEqual(err.code, 'ECONNRESET');
```

Comment:
I think `err.code` can also be `'EPIPE'` or `'ECONNABORTED'` but I was not able to reproduce. Should we handle that in advance or do it if/when it will occur?

Reply:
I'd say let's worry about it when it occurs.

Reply:
Perhaps you can add a comment right now that says "EPIPE might be valid too but we haven't seen it yet" or something like that?